### DISCUSSION TITLE
Does Unit support Symfony's kernel.terminate?

### DISCUSSION THREAD

**Discussion by p4block** [nginx/unit#1002] [Support]
Migrated my app from nginx+php-fpm to Unit. We're using Sentry for profiling and error handling.
There's this bit: 

> If your web server is using FastCGI, our SDK utilizes the [kernel.terminate event](https://symfony.com/doc/current/components/http_kernel.html#8-the-kernel-terminate-event)
> 
> , which means the response is sent to the user before the performance data is sent to Sentry.
> 
> If your web server isn't using FastCGI, you can run [Relay](https://docs.sentry.io/product/relay/getting-started/?original_referrer=https%3A%2F%2Fdocs.sentry.io%2Fplatforms%2Fphp%2Fguides%2Fsymfony%2Fusage%2Fdistributed-tracing%2F) locally on the same machine or a local network that can act as a proxy/agent.
> 
> Doing this will make the PHP process send requests to your local Relay, which will then forward them to Sentry, instead of sending requests to Sentry directly.

**Comment by ac000** [comment]
FWIW.

I tried the first test program from [here](https://www.php.net/manual/en/function.fastcgi-finish-request.php) and it did indeed create the file.

```shell
$ curl localhost:8080/fastcgi_finish_request.php
<p>Calling to <code>fastcgi_finish_request()</code>.</p>
<p>If success, the file /home/andrew/src/php/text.txt will be created.</p>
$ cat /home/andrew/src/php/text.txt
2023-11-10 16:28:09
```

That suggests that `fastcgi_finish_request()` worked which is seemingly what is used by `kernel.terminate`.

So it looks like it at least has a chance of working. I guess your best bet is to just try it...

Feel free to report back your findings. If it doesn't work we can look at what needs fixing in Unit.

**Comment by p4block** [comment]
Well, I'm a big dummy. The actual "problematic" php call it runs is indeed [fastcgi_finish_request](https://secure.php.net/manual/en/function.fastcgi-finish-request.php)

Which is **specifically mentioned** in the basic php docs as supported! 
https://unit.nginx.org/configuration/#configuration-php

Next week I am tasked with intense performance testing and I will check if it works as advertised, but there's no reason it shouldn't.
